cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/store/DatasetBasedTimeScheduleStore.java

```diff
@@ -104,7 +104,7 @@ public void storeJobAndTrigger(JobDetail newJob, OperableTrigger newTrigger)
     super.storeJob(newJob, true);
     super.storeTrigger(newTrigger, true);
 
-    executePersist(newJob, newTrigger);
+    executePersist(newJob, newTrigger, super.getTriggerState(newTrigger.getKey()));
```

Comment:
I think this comment is for storeJobsAndTriggers() and github is collapsing the lines.
Yes that is what I am planning to do. Thats how super.storeJobsAndTriggers() does too with some validation initially will use the same logic.